swiper.el

```diff
@@ -503,6 +503,17 @@ When non-nil, INITIAL-INPUT is the initial search pattern."
   (interactive)
   (swiper--ivy (swiper--candidates) initial-input))
 
+;;;###autoload
+(defun swiper-region-or-not ()
+  "If region is selected, `swiper' with the keyword selected in region.
+If the region isn't selected, `swiper'."
+  (interactive)
+  (if (region-active-p)
+      (progn (setq mark-active nil)
```

Comment:
You should usually call the function `deactivate-mark` or set the variable `deactivate-mark` instead of manipulating `mark-active` directly; see [`(elisp) The Mark`](https://www.gnu.org/software/emacs/manual/html_node/elisp/The-Mark.html) for details.

Reply:
Fix to deactivate-mark.